src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_utils.py

```diff
@@ -37,3 +39,87 @@ def create_self_signed_certificate(device_id, valid_days, cert_output_dir):
         'privateKey': key_dump,
         'thumbprint': thumbprint
     }
+
+
+def open_certificate(certificate_path):
+    certificate = ""
+    if certificate_path.endswith('.pem') or certificate_path.endswith('.cer'):
+        with open(certificate_path, "rb") as cert_file:
+            certificate = cert_file.read()
+            try:
+                certificate = certificate.decode("utf-8")
+            except UnicodeError:
+                certificate = base64.b64encode(certificate).decode("utf-8")
```

Comment:
Just to confirm, we don't want to encode the certificate with base64 if all the characters are valid unicode?

Reply:
Yes, that's correct.